Change loading of OVAL result reports

Honny1 opened the pull request:
This PR changes the approach of loading the OVAL results report and brings special logic to determine where is present the OVAL result report with the OVAL definitions that are used in the CPE dictionary and CPE AL. 

The OVAL result report element is selected according to the references in the rule results. The selected element is the one that is not referenced in the rule results. If multiple report elements are not referenced in the rule result, OVAL definitions for CPE are selected from all report elements. In the case of duplicate definitions, the first definition or the definition that has a result other than `not evaluated` is chosen.

evgenyz commented on openscap_report/scap_results_parser/parsers/rule_parser.py:
```diff
@@ -146,6 +146,16 @@ def _add_message_about_oval(rule_id, rules):
+            return None
+        check_name = check_ref.get("name")
+        if rule.oval_definition_id is None:
+            rule.oval_definition_id = check_name
```
The method is a 'getter', but it *sometimes* mutates the object. On top of that, it is a **static** *private* method. That's very confusing.

evgenyz commented on openscap_report/scap_results_parser/scap_results_parser.py:
```diff
@@ -75,6 +75,14 @@ def _get_benchmark_element(self):
         return benchmark_el
 
+    @staticmethod
+    def _get_oval_definition_reference(rules):
```
That should be either `references` or `reference_list`.

evgenyz commented on openscap_report/scap_results_parser/scap_results_parser.py:
```diff
@@ -88,10 +96,11 @@ def parse_report(self):
         OVAL_and_CPE_tree_builder = OVALAndCPETreeBuilder(  # pylint: disable=C0103
             self.root, group_parser,
-            self._get_applicable_cpe_ids_for_machine(report.profile_info.cpe_platforms_for_profile)
+            self._get_applicable_cpe_ids_for_machine(report.profile_info.cpe_platforms_for_profile),
```
The `ProfileInfo` class already has `get_cpe_platforms_that_satisfy_evaluation_target` getter, which kinda does the same under a different name. This is inconsistent.

I'd say that `_get_applicable_cpe_ids_for_machine` should be part of `ProfileInfo` (named like `get_list_of_cpe_platforms_that_satisfy_evaluation_target`) and `get_cpe_platforms_that_satisfy_evaluation_target` should use it to join a list into a string.

evgenyz commented on openscap_report/scap_results_parser/oval_and_cpe_tree_builder.py:
```diff
@@ -9,53 +9,89 @@
+            return self.reports_with_oval_definitions[self.cpe_source]
+        logging.warning((
+            "The given input does not contain a clear mapping of the OVAL definition used"
+            " for CPE checks. The results of the OVAL definition in the CPE checks could "
```
Make the way you split strings consistent: space either goes to the end or to the start of the line.

evgenyz commented on openscap_report/scap_results_parser/oval_and_cpe_tree_builder.py:
```diff
@@ -98,13 +134,29 @@ def _remove_double_cpe_requirement(rule, group_platforms):
+            if oval_def is not None and rule.oval_definition_id in report:
+                logging.warning(
+                    ("The given input contains the duplicate results of"
+                     " the OVAL definition (\"%s\")."),
```
Make the way you split strings consistent: space either goes to the end or to the start of the line.

Honny1 commented:
Changes:
- Consistently split strings
- Reworked _get_oval_check_href method
- Renamed _get_oval_definition_reference method to _get_oval_definition_references
- The _get_applicable_cpe_ids_for_machine method has been moved to ProfileInfo
- Renamed _get_applicable_cpe_ids_for_machine method to get_list_of_cpe_platforms_that_satisfy_evaluation_target
